nexus/src/app/sagas/common_storage.rs

```diff
@@ -37,10 +38,42 @@ pub(crate) async fn get_pantry_address(
     Ok(client.address())
 }
 
+// Helper function for attach/detach below: we retry as long as the pantry isn't
+// gone, and we detect "gone" by seeing whether the pantry address we've chosen
+// is still present when we resolve all the crucible pantry records in DNS.
+//
+// This function never returns an error because it's expected to be used with
+// `ProgenitorOperationRetry`, which treats an error in the "gone check" as a
+// fatal error. We don't want to amplify failures: if something is wrong with
+// DNS, we can't go back and choose another pantry anyway, so we'll just keep
+// retrying until DNS comes back. All that to say: a failure to resolve DNS is
+// treated as "the pantry is not gone".
+pub(super) async fn is_pantry_gone(
+    nexus: &Nexus,
+    pantry_address: SocketAddrV6,
+    log: &Logger,
+) -> bool {
+    let all_pantry_dns_entries = match nexus
+        .resolver()
+        .lookup_all_socket_v6(ServiceName::CruciblePantry)
+        .await
+    {
+        Ok(entries) => entries,
+        Err(err) => {
+            warn!(
+                log, "Failed to resolve Crucible pantry in DNS";
```

Comment:
```suggestion
                log, "Failed to query DNS for Crucible pantry";
```

super nitty but sometimes "failed to resolve X in DNS" means "I successfully reached DNS and X wasn't there", which isn't the case here

Reply:
Done in [fb21fb9](https://github.com/oxidecomputer/omicron/pull/6866/commits/fb21fb945740a624a0d93ddeda5970f683bd43a1)